src/components/Logout/LogoutScreen.test.tsx

```diff
@@ -0,0 +1,298 @@
+    expect(mockCallLogout).toHaveBeenCalledWith("sesA", "optA", "endA");
+    expect(mockCallLogout).toHaveBeenCalledWith("sesB", "optB", "endB");
+    expect(mockCallLogout).toHaveBeenCalledWith("sesC", "optC", "endC");
+  });
+
+  it("should logout from campaigns with no error and vice versa", async () => {
+    expect.assertions(9);
+
+    mockCallLogout.mockImplementation(async (sessionToken) => {
+      if (sessionToken === "sesA") throw new LogoutError("");
+    });
+    const mockRemoveAuthCredentials = jest
+      .fn()
+      .mockName("removeAuthCredentials");
+    const mockRemoveCampaignConfig = jest.fn().mockName("removeCampaignConfig");
+
+    render(
+      <AuthStoreContext.Provider
+        value={{
+          ...defaultAuthStoreContextValue,
+          removeAuthCredentials: mockRemoveAuthCredentials,
+        }}
+      >
+        <CampaignConfigsStoreContext.Provider
+          value={{
+            ...defaultCampaignConfigStoreContextValue,
+            removeCampaignConfig: mockRemoveCampaignConfig,
+          }}
+        >
+          <LogoutScreenContainer />
+        </CampaignConfigsStoreContext.Provider>
+      </AuthStoreContext.Provider>
+    );
+
+    await waitFor(() => expect(mockNavigate).toHaveBeenCalledTimes(1));
+
+    expect(mockRemoveAuthCredentials).toHaveBeenCalledTimes(2);
+    expect(mockRemoveAuthCredentials).not.toHaveBeenCalledWith(keyA);
+    expect(mockRemoveAuthCredentials).toHaveBeenCalledWith(keyB);
+    expect(mockRemoveAuthCredentials).toHaveBeenCalledWith(keyC);
+
+    expect(mockRemoveCampaignConfig).toHaveBeenCalledTimes(2);
+    expect(mockRemoveCampaignConfig).not.toHaveBeenCalledWith(keyA);
+    expect(mockRemoveCampaignConfig).toHaveBeenCalledWith(keyB);
+    expect(mockRemoveCampaignConfig).toHaveBeenCalledWith(keyC);
+  });
+
+  it("should clear all credentials and configs and move to login screen upon successful logout", async () => {
+    expect.assertions(4);
+
+    const mockClearAuthCredentials = jest.fn();
+    const mockClearCampaignConfigs = jest.fn();
+
+    render(
+      <AuthStoreContext.Provider
+        value={{
+          ...defaultAuthStoreContextValue,
+          clearAuthCredentials: mockClearAuthCredentials,
+        }}
+      >
+        <CampaignConfigsStoreContext.Provider
+          value={{
+            ...defaultCampaignConfigStoreContextValue,
+            clearCampaignConfigs: mockClearCampaignConfigs,
+          }}
+        >
+          <LogoutScreenContainer />
+        </CampaignConfigsStoreContext.Provider>
+      </AuthStoreContext.Provider>
+    );
+
+    await waitFor(() => expect(mockNavigate).toHaveBeenCalledTimes(1));
+
+    expect(mockNavigate).toHaveBeenCalledWith("LoginScreen");
+    expect(mockClearAuthCredentials).toHaveBeenCalledTimes(1);
+    expect(mockClearCampaignConfigs).toHaveBeenCalledTimes(1);
+  });
+
+  it("should not clear all data and should move to campaign selection screen upon unsuccessful logout", async () => {
```

Comment:
Hmm is this when there is 1 single campaign or multiple campaigns? Is it an extension of the second `it` block above?

Reply:
If you find that you are setting up for the test case the same way in many tests, you can reorganise your tests into this structure

```
describe(‘name of tests’, () => {
  describe(‘when there are multiple campaigns’, () => {
      describe(‘when logout for all campaigns is successful’, () => {
        beforeAll(() => { 
          // set up for test here once! 
        });

        it(‘should clear all data’, () => { 
          // …
      });

      describe(‘when logout for one of the campaigns failed’, () => {
        beforeAll(() => { 
          // set up for test here once! 
        });
        it(‘should do something about state’, () => {…});
        it(‘should show some alert’, () => {…});
      });
  });

  describe(‘when there is 1 campaign’, () => {…});
});
```

This way it’s easier to understand your tests because they are structured by scenarios. And you don’t have so much duplicated code trying to set up the tests the same way every time. Can consider~

Reply:
This should be true regardless of whether or not there is one campaign or many. Currently, the many campaign case is the default and there's no special treatment for that so I didn't test it. The special case is when there is only one campaign left after logout which is what is being tested here (`campaign A` is the one that is remaining here)